plugins/inputs/net/net.go

```diff
@@ -129,6 +133,25 @@ func (n *NetIOStats) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
+// Get the interface speed from /sys/class/net/*/speed file. returns -1 if unsupported
+func getInterfaceSpeed(ioName string) int64 {
+	sysPath := os.Getenv("HOST_SYS")
+	if sysPath == "" {
+		sysPath = "/sys"
+	}
+
+	raw, err := os.ReadFile(filepath.Join(sysPath, "class", "net", ioName, "speed"))
+	if err != nil {
+		return -1
+	}
+
+	speed, err := strconv.ParseInt(strings.TrimSuffix(string(raw), "\n"), 10, 64)
+	if err != nil {
+		return -1
+	}
```

Comment:
Should we really return `-1` or better return `nil` or not add the field at all?

Reply:
I'm not sure. I think for the file read, -1 makes sense since this file can be absent indicating that its unsupported, similar to reporting -1. For the ParseInt, I'm more unsure but thought it was better to have one "failed" case in the reported metrics than multiple

Reply:
Yeah but why not leave out the field completely if it does not contain any real data?

Reply:
You would potentially have two different series produced. I also would want to avoid having people go why does this show up for some nics, but not others.

Reply:
Makes sense. Then I'm fine.